File: Lb5_cpp/gui/MainWindow.cpp
#include "MainWindow.h"

MainWindow::MainWindow(QWidget *parent) : QWidget(parent) {
    this->setWindowTitle("Nechepurenko & Terekhov Ltd.");
    this->setFixedSize(800, 480);
    this->mainLayout = new QHBoxLayout();
    this->setLayout(this->mainLayout);
    this->controlPanel = new ControlPanel();
    this->drawArea = new DrawArea(this->controlPanel);
    this->controlPanel->setDrawArea(this->drawArea);
    this->mainLayout->addWidget(this->drawArea);
    this->mainLayout->addWidget(this->controlPanel);
}

MainWindow::~MainWindow() {
    delete this->mainLayout;
    delete this->drawArea;
    delete this->controlPanel;
}

File: Lb6/gui/MainWindow.py
from PyQt5.QtWidgets import QWidget, QHBoxLayout

from gui.drawarea import DrawArea
from gui.panel import ControlPanel
from utils.mediator import Mediator


class MainWindow(QWidget):
    def __init__(self):
        super().__init__()
        self.setWindowTitle("Nechepurenko & Terekhov Ltd.")
        self._main_layout = QHBoxLayout()
        self.setLayout(self._main_layout)
        mediator = Mediator()
        self._draw_area = DrawArea(mediator)
        self._main_layout.addWidget(self._draw_area)
        self._control_panel = ControlPanel(self._draw_area)
        self._main_layout.addWidget(self._control_panel)
        self.setMaximumSize(700, 480)
